backends/eopkg: Use the privileged decorator to deduplicate common code

joebonrichie opened the pull request:
Partially resolves #11.

joebonrichie committed:
backends/eopkg: Use the privileged decorator to deduplicate common code

EbonJaeger commented on backends/eopkg/eopkgBackend.py:
```diff
@@ -113,17 +113,61 @@ def _load_settings(self):
+
+    def status_cb(self, event, **keywords):
+        # FIXME: when getting repo, we can't use self.packagedb here as it interferes with atomicoperations
+        if event == pisi.ui.downloading:
+            self.status(STATUS_DOWNLOAD)
+            pkg = keywords["package"]
+            if self.packagedb.has_package(pkg.name):
+                repo = pisi.db.packagedb.PackageDB().which_repo(pkg.name)
+            else:
+                repo = "local"
+            pkg_id = self.get_package_id(pkg.name, pkg.version, pkg.architecture, repo)
+            self.package(pkg_id, INFO_DOWNLOADING, pkg.summary)
+        if event == pisi.ui.installing:
+            self.status(STATUS_INSTALL)
+            pkg = keywords["package"]
+            if self.packagedb.has_package(pkg.name):
+                repo = pisi.db.packagedb.PackageDB().which_repo(pkg.name)
+            else:
+                repo = "local"
+            pkg_id = self.get_package_id(pkg.name, pkg.version, pkg.architecture, repo)
+            self.package(pkg_id, INFO_INSTALLING, pkg.summary)
+        if event == pisi.ui.removing:
+            self.status(STATUS_REMOVE)
+            pkg = keywords["package"]
+            if self.packagedb.has_package(pkg.name):
+                repo = pisi.db.packagedb.PackageDB().which_repo(pkg.name)
+            else:
+                repo = "installed"
+            pkg_id = self.get_package_id(pkg.name, pkg.version, pkg.architecture, repo)
+            self.package(pkg_id, INFO_REMOVING, pkg.summary)
```
Maybe I'm not seeing something obvious, but this looks like a bunch of duplicated code. Can we do only the event-specific operations in the if/else, and the common operations outside of the if/else?

joebonrichie replied:
I thought the same but the event needs to be non null and the repo status can be slightly different for local pkgs for different events which in turn leads up up to it not really saving that many loc. If I do find a nice way I'll implement it.

joebonrichie committed:
backends/eopkg: Progress bar / status crap hooked up again

joebonrichie committed:
backends/eopkg: Handle common exceptions for common operations

Any issues will now will be more visible to the user.

joebonrichie committed:
backends/eopkg: Catch removing system.base pkgs during simulate state

Catch the issue up front.

joebonrichie committed:
backends/eopkg: Support simulating install_files

This now works as expected. Extra work may need to be done to resolve
conflicts and replaces but the same is true across the board.

joebonrichie committed:
backends/eopkg: Implement repair_system support
